test/common/buffer/watermark_buffer_test.cc

```diff
@@ -430,23 +430,34 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
                                   [&]() -> void { ++overflow_watermark_buffer1; }};
 
   // Make sure the overflow threshold will be above std::numeric_limits<uint32_t>::max()
-  Runtime::LoaderSingleton::getExisting()->mergeValues({{"envoy.buffer.overflow_multiplier", "3"}});
-  buffer1.setWatermarks((std::numeric_limits<uint32_t>::max() / 3) + 1);
-
-  // Add many segments instead of full uint32_t::max to get around std::bad_alloc exception
-  const uint32_t segment_denominator = 128;
-  const uint32_t big_segment_len = std::numeric_limits<uint32_t>::max() / segment_denominator + 1;
-  const std::string big_segment_str = std::string(big_segment_len, 'a');
-  for (uint32_t i = 0; i < segment_denominator; ++i) {
-    buffer1.add(big_segment_str.data(), big_segment_len);
-  }
+  const uint64_t overflow_multiplier = 3;
+  Runtime::LoaderSingleton::getExisting()->mergeValues(
+      {{"envoy.buffer.overflow_multiplier", std::to_string(overflow_multiplier)}});
+  const uint32_t high_watermark_threshold =
+      (std::numeric_limits<uint32_t>::max() / overflow_multiplier) + 1;
+  buffer1.setWatermarks(high_watermark_threshold);
+
+  Buffer::RawSlice iovec;
+  // Reserve and commit uint32_t::max + 1 bytes
+  uint64_t num_reserved =
+      buffer1.reserve(std::numeric_limits<uint32_t>::max() + UINT64_C(1), &iovec, 1);
```

Comment:
You're effectively doing a single allocation of uint32_t max, this will break in cases the allocator is configured to complain about attempts to allocate buffers that are beyond some thresholds.  You need to preserve the big_segment_len logic, do reserve and commit in a loop.

Reply:
Done.